pkg/master/controller.go

```diff
@@ -132,10 +132,23 @@ func (c *Controller) UpdateKubernetesService(reconcile bool) error {
 	return nil
 }
 
-// CreateNamespaceIfNeeded will create the namespace that contains the master services if it doesn't already exist
+// EnsureSystemNamespacesExist will make sure both the default and the system namespace always will exist
+func (c *Controller) EnsureSystemNamespacesExist() error {
+	// Try to create the default namespace
+	if err := c.CreateNamespaceIfNeeded(api.NamespaceDefault); err != nil {
```

Comment:
I was thinking about something like this also:
```go
c.SystemNamespacesInterval = 1 * time.Minute
c.SystemNamespaces = []string {api.NamespaceSystem}

// RunKubernetesNamespaces periodically makes sure that all internal namespaces exist
func (c *Controller) RunKubernetesNamespaces(ch chan struct{}) {
	wait.Until(func() {

		// Loop the system namespace list, and create them if they do not exist
		for _, ns := range c.SystemNamespaces {

			if err := c.CreateNamespaceIfNeeded(ns); err != nil {
				runtime.HandleError(fmt.Errorf("unable to create required kubernetes system namespace %s: %v", ns, err))
			}
		}
	}, c.SystemNamespacesInterval, ch)
}
```
Do you think we should go with that approach instead?
It's more dynamic when it's only a list to modify.
The `default` ns will be created at the service creation anyway, so there's no need to add it to the list